src/storm/generator/JaniNextStateGenerator.cpp

```diff
@@ -68,6 +68,8 @@ JaniNextStateGenerator<ValueType, StateType>::JaniNextStateGenerator(storm::jani
     }
     STORM_LOG_THROW(features.empty(), storm::exceptions::NotSupportedException,
                     "The explicit next-state generator does not support the following model feature(s): " << features.toString() << ".");
+    // Simplify the system compositions so that we can exclude the case where automata appear in the composition multiple times.
```

Comment:
So simplify also brings it in some kind of normal form that is necessary?  The name "simplify" may be a bit misleading, but the comment is sufficient :-)

Reply:
One could argue that simplify brings it to a *simple form* that is necessary. But I agree that `simplifyComposition` could be renamed to something more descriptive.